model-optimizer/extensions/back/ReverseInputChannels.py

```diff
@@ -295,38 +295,41 @@ class ReverseChannelsPropagationUp(BackReplacementPattern):
         'Subtract': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
         'Pow': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
         'Convert': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
-
-        'Pad': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through(node, rc),
+        'Pad': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_pad(node, rc),
     }
 
     @staticmethod
-    def lift_up_through(node: Node, reverse_channels: Node):
+    def lift_up_through_pad(node: Node, reverse_channels: Node):
         r"""
         BEFORE                       AFTER
 
                                      previous_op
                                           \
         previous_op  previous_op       ReverseChannels  previous_op
                  \     /                           \     /
-                   Node                             Node
+                   Pad                              Pad
                     |                                |
               ReverseChannels                      next_op
                     |
                  next_op
 
-        returns boolean value whatever we should continue propagating current ReverseChannels operation up or not
+        returns two objects:
+        first - boolean value whatever we should continue propagating current ReverseChannels operation up or not
+        second - list of ReverseChannels operations that were produced while propagating reverse_channels up
         """
         if node.is_in_port_connected(0):
             node_input_port_0 = node.in_port(0)
-            reverse_channels_out_npde = reverse_channels.out_port(0).get_connection().get_destination().node
+            reverse_channels_out_nodes = [destination.node for destination in
+                                          reverse_channels.out_port(0).get_connection().get_destinations()]
             reverse_channels.out_port(0).disconnect()
 
             src = node_input_port_0.get_connection().get_source()
             node_input_port_0.get_connection().set_source(reverse_channels.out_port(0))
             src.connect(reverse_channels.in_port(0))
-            node.out_port(0).get_connection().set_destination(reverse_channels_out_npde.in_port(0))
-            return True
-        return False
+            for reverse_channels_out_node in reverse_channels_out_nodes:
+                node.out_port(0).get_connection().set_destination(reverse_channels_out_node.in_port(0))
```

Comment:
`set_destination` sets only destination for connection. you need to use `add_destination`.
Please add unit-test for this case.